### ISSUE TITLE
Define a one-dimensional (spectral fit) statistical model suitable for low/high energy

### ISSUE THREAD

**Issue by kdund** [XENONnT/alea#92]
Several searches use 1D spectral fits, binned or unbinned. 
They use sometimes functional forms for background/signals, and sometimes templates. 
It would be great to make a general class for these searches! 

Such a class should: 
- Allow the definition of binned or unbinned likelihoods
- Allow the definition of sources as: 
  - Functional forms (in particular ease of putting in flat+gammas) 
  - Templates (using inference interface) 
  - with rate uncertainties and ancillary constraints as in the blueice class. 
  - With or without applying an efficiency curve + convolution either in  true energy (standard convolution) or in reconstructed energy (as in the lowER search) 
- Allow combined fits of several 1D fits
- Allow useful goodness of fit tests?

**Comment by kdund** [comment]
@cecilia-ferrari @hammannr since we chatted.

**Comment by hammannr** [comment]
Also @matteoguida might be interested

**Comment by matteoguida** [comment]
@kdund, @hammannr, I'll try to contribute to developing this class, though it might not be in the immediate future.

**Comment by jingqiangye** [comment]
Yeah that'll be cool! In SR0 LowER search, I had to define different classes for each of them -- flat, gaussian, skew-guassian, and template-based. One additional thing to add on top of your list of requirements -- allow to change the region of interest, for example in the case of blinding or just the need to optimize.

**Comment by cecilia-ferrari** [comment]
Ciao!
Based on the code @GiovanniVolta  developed for Nucleon Decay disappearance studies, we built a tool for 1D CES fit. It is currently called [he_er](https://github.com/XENONnT/nton/tree/he_er/nton/he_er) but as shown at the end of this [example notebook](https://github.com/XENONnT/analysiscode/blob/c4d58aa6f110a4e7598cc4a37f41a7e5cd30651d/he_er/he_er_example.ipynb) can also perform analysis in the low er ROI. Also in this case, depending on the nature of the background component (peak, theoretical spectrum or G4 generated one) the model is implemented with a specific class.
As discussed in the [pull request thread](https://github.com/XENONnT/nton/pull/179#issuecomment-1587664566), some features still need improvements.
